src/plugins/intel_gpu/src/kernel_selector/kernels/reorder/reorder_weights_int4.cpp

```diff
@@ -50,24 +50,82 @@ ReorderWeightsKernelInt4::DispatchData ReorderWeightsKernelInt4::SetDefault(cons
         dispatchData.gws = { Align(output.OFM().v, 64) / 2, output.IFM().v, 1 };
     } else if (output.GetLayout() == WeightsLayout::os_is_yx_osv64_isv2) {
         dispatchData.gws = { Align(output.OFM().v, 64), output.IFM().v / 2, 1 };
-    } else {
-        dispatchData.gws = { CeilDiv(output.LogicalSize(), 2), 1, 1 };
+    } else if (output.GetLayout() == WeightsLayout::oiyx) {
+        auto dims = output.GetDims();
+        bool has_pads = std::any_of(dims.begin(), dims.end(), [](const kernel_selector::Tensor::Dim& d) {
+            return d.pad.Total() != 0;
+        });
+        if (has_pads) {
+            dispatchData.gws = { CeilDiv(output.PhysicalSize(), 2), 1, 1 };
+        } else {
+            dispatchData.gws = { CeilDiv(output.LogicalSize(), 2), 1, 1 };
+        }
     }
     dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
 
     return dispatchData;
 }
 
+static inline  kernel_selector::Tensor::NDims GetTrimmedDims(const kernel_selector::Tensor::NDims& org_dims) {
+    auto it = org_dims.begin();
+    while (it != org_dims.end() && it->v == 1) {
+        ++it;
+    }
```

Comment:
Same question, is this for the legacy cldnn shapes only?

Reply:
Yes

Reply:
Modified the code as discussed offline. thanks.